code/integration-tests/runtime-tests/tests/utils/stakingUtils.ts

```diff
@@ -0,0 +1,160 @@
+import {ApiPromise, Keyring} from "@polkadot/api";
+import {KeyringPair} from "@polkadot/keyring/types";
+import BigNumber from "bignumber.js";
+import {sendAndWaitForSuccess} from "./txClient";
+import {stakingPalletAddress} from "./constants";
+import {waitForSeconds} from "./ibcUtils";
+
+
+export async function seedStakingPallet(api: ApiPromise, sudoKey: KeyringPair, tokenId: number, transferAmount: BigNumber) {
+  await sendTokens(api, sudoKey, stakingPalletAddress, tokenId, false, transferAmount);
+}
+
+export async function sendTokens(
+  api: ApiPromise,
+  senderWallet: KeyringPair,
+  receiverWalletAddress: string,
+  tokenId: number,
+  expectedToFail: boolean,
+  transferAmount: BigNumber
+) {
+  const asset = api.createType('u128', tokenId);
+  const dest = api.createType('MultiAddress', {
+    id: api.createType('AccountId', receiverWalletAddress)
+  });
+  const amount = api.createType('Compact<u128>', transferAmount.toString());
+  const keepAlive = api.createType('bool', 'No');
+  let event;
+  if (expectedToFail) {
+    event = api.events.system.ExtrinsicFailed.is;
+  } else {
+    event = api.events.system.ExtrinsicSuccess.is;
+  }
+  return await sendAndWaitForSuccess(
+    api,
+    senderWallet,
+    // @ts-ignore
+    event,
+    api.tx.assets.transfer(asset, dest, amount, keepAlive),
+    expectedToFail
+  )
+}
+
+export async function createStakingPool(
+  api: ApiPromise,
+  sudoKey: KeyringPair,
+  stakingTokenId: number,
+  rewardAmount: BigNumber,
+  rewardTokenId: number,
+  period: number
+) {
+  const poolCurrencyId = api.createType('u128', stakingTokenId);
+  const rewardCurrencyId = api.createType('u128', rewardTokenId);
+  const periodCount = api.createType('u32', period);
+  const amount = api.createType('u128', rewardAmount.toString());
+  return await sendAndWaitForSuccess(
+    api,
+    sudoKey,
+    api.events.farming.RewardScheduleUpdated.is,
+    api.tx.sudo.sudo(api.tx.farming.updateRewardSchedule(poolCurrencyId, rewardCurrencyId, periodCount, amount)),
```

Comment:
jfy @dzmitry-lahoda
in current version of pallet implementation it is not correct to update reward schedule if schedule started already and there is funds on pool account that user did not claimed yet.
here is the [pallet code](https://github.com/ComposableFi/composable/blob/main/code/parachain/frame/farming/src/lib.rs#L271-L272) that you need to be aware

so the actual schedule update supposed to be this steps
- remove current scheduled
- create the new one schedule
- transfer to the the pool not claimed reward from prev schedule 

In other case you should not call update reward schedule extrinsic.

to allow safely update existing reward pool on the fly we need to modify this [line of code](https://github.com/ComposableFi/composable/blob/main/code/parachain/frame/farming/src/lib.rs#L271-L272)

and recalculate next schedule reward based on current schedule that not finished yet.